minifi/minifi-commons/minifi-utils/src/main/java/org/apache/nifi/minifi/commons/status/util/PropertyUtil.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.commons.status.util;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class PropertyUtil {
+    private static final String DOT = ".";
+    private static final String UNDERSCORE = "_";
+    private static final String HYPHEN = "-";
+
+    private PropertyUtil() {
+
+    }
+
+    public static String resolvePropertyValue(String name, Map<?, ?> properties) {
+        return Optional.of(name)
+            .map(n -> getPropertyValue(n, properties))
+            .or(() -> Optional.of(name.toUpperCase())
```

Comment:
There is a lot happening here, wouldn't it be cleaner to do uppercase for both name and the properties keys? I know  it would mean extra iteration over properties but would make the code cleaner in my opinion. What do you think?

Reply:
discussed it in a separate channel. As the utility approach does not allow this easily we leave it as it is.